pinot-core/src/main/java/org/apache/pinot/core/segment/processing/mapper/SegmentMapper.java

```diff
@@ -136,6 +136,7 @@ public Map<String, GenericRowFileManager> map()
 
         reuse.clear();
       }
+      recordReader.close();
```

Comment:
Can we release the reader resources here, after we are done processing the file ?